File: go/648.go
package main

import (
	"strings"
)

func replaceWords(dictionary []string, sentence string) string {
	tokens := strings.Split(sentence, " ")

	ret := ""

	for i := 0; i < len(tokens); i++ {
		for j := 0; j < len(dictionary); j++ {
			if strings.HasPrefix(tokens[i], dictionary[j]) {
				tokens[i] = dictionary[j]
			}
		}
		if len(ret) != 0 {
			ret += " "
		}
		ret += tokens[i]
	}

	return ret
}

File: cpp/648.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  string replaceWords(vector<string> &d, string s) {
    char *cs = new char[s.length() + 1];
    cs[s.length()] = '\0';
    for (int i = 0; i < s.length(); i++)
      cs[i] = s[i];
    vector<string> tokens;
    for (char *tok = strtok(cs, " "); tok != nullptr;
         tok = strtok(nullptr, " "))
      tokens.push_back(string(tok));

    string ret = "";
    for (int i = 0; i < tokens.size(); i++) {
      for (int j = 0; j < d.size(); j++) {
        auto re = mismatch(d[j].begin(), d[j].end(), tokens[i].begin());
        if (re.first == d[j].end())
          tokens[i] = d[j];
      }
    }
    for (int i = 0; i < tokens.size(); i++) {
      if (ret.size() != 0)
        ret += " ";
      ret += tokens[i];
    }
    return ret;
  }
};
